docs/docs/dev-manuals/services/frontend.mdx

```diff
@@ -0,0 +1,161 @@
+ */
+const client = useClient()
+
+/**
+ * Gets the "entire" product from the catalog service.
+ */
+const product = asyncComputed(
+    async () => {
+        return client.getProduct({
+            id: id.value,
+        })
+    },
+    null,
+    {
+        onError: (e) => pushErrorNotification(errorMessages.getProduct, e),
+        shallow: false,
+    }
+)
+```
+
+Note that the GraphQL query "behind" `getProduct()` has been defined in `src/graphql/catalog.graphql`.
+
+### The error handler
+
+`src/util/errorHandler.ts` provides intuitive helper functions for wrapping synchronous as well as asynchronous actions into `try-catch` blocks in order to push an error notification if the actions fails.
+
+```jsx
+/**
+ * Requests the restocking of the selected product variant.
+ */
+async function restock() {
+    if (activeUserRoleIsEitherAdminOrEmployee.value && canRestock) {
+        await awaitActionAndPushErrorIfNecessary(() => {
+            return client.createProductItemBatch({
+                input: {
+                    productVariantId: selectedProductVariantId.value,
+                    number: parseInt(amount.value as string),
+                },
+            })
+        }, errorMessages.createProductItemBatch)
+
+        emit('restocked')
+    }
+}
+```
+
+In this code example `awaitActionAndPushErrorIfNecessary()` (one of the error handler's functions) is called with two arguments:
+
+- the asynchronous action
+- the error message if the action fails: errorMessages.createProductItemBatch
+
+**Whenever the failure of a function call should be communicated to the user, use one of the error handler's functions.**
+
+## Technical Debt
+
+:::info
+This section is written from the personal perspective of the main frontend developer during the time of the EnPro.
+:::
+Sadly, there is already some technical debt. Why? Because at the beginning of the project I was still a complete beginner when it came to frontend development with Vue, TypeScript, and JavaScript in general. Some things I only understood and learned to use properly over time meaning issue by issue. As we did not have the time to refactor everything as soon as possible, we had to prioritize the implementation of additional features to ensure that the frontend would be at least somehow feature-complete at the end of the project.
+
+Having said this, what kind of technical debt are we talking about?
+
+- Clones regarding the composition of smaller UI components. That means more custom SFCs could be extracted from the given DOM landscape.
+- A slight overuse of `v-if` within SFCs instead of making good use of Vue's slot mechanism. See [this Gropius issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/4f780153-0e6a-4893-a3b9-4e019b74a2a8).
+- Using events instead of custom models. See [this Gropius issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/4f780153-0e6a-4893-a3b9-4e019b74a2a8).
```

Comment:
I would not rely on Gropius being available forever (especially not under this domain).
Perhaps you can add an excerpt of the issue like so
```md
- Using events instead of custom models. See [this Gropius issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/4f780153-0e6a-4893-a3b9-4e019b74a2a8).
If this link no longer works,
<details><summary>Here is the relevant part</summary>

…
</details>
```
?

Reply:
Then again, it's only an example of technical debt and this particular one will hopefully be fixed within a few weeks. I would like to just keep it the way it is.